observability/src/main/java/io/grpc/observability/interceptors/InternalLoggingChannelInterceptor.java

```diff
@@ -100,12 +88,14 @@ private InternalLoggingChannelInterceptor(LogHelper helper) {
     final Deadline deadline = LogHelper.min(callOptions.getDeadline(),
         Context.current().getDeadline());
 
-    // TODO (dnvindhya): implement isMethodToBeLogged() to check for methods to be logged
-    // according to config. Until then always return true.
-    if (!helper.isMethodToBeLogged(method.getFullMethodName())) {
+    if (!filterHelper.isMethodToBeLogged(method.getFullMethodName()).log) {
       return next.newCall(method, callOptions);
     }
 
+    MethodFilterParams filterParams = filterHelper.isMethodToBeLogged(method.getFullMethodName());
```

Comment:
if `filterParams.log` is false you can just return similar to line 92 right?

Reply:
If `filterParams.log` is false the function returns at L92 and L95 will not be executed in that case. So did not add that check again.